src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -76,4 +94,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Gson gson = new Gson();
     response.getWriter().println(gson.toJson(loginStatus));
   }
+
+  public String generateFriendCode() {
+    try {
+      String values = "0123456789abcdefghijklmnopqrstuvwxyz";
+      SecureRandom secureRandom = SecureRandom.getInstanceStrong();
+
+      // Generate a string of length 12.
+      String randomString =
+          secureRandom
+              .ints(12, 0, values.length())
```

Comment:
nit: it might be better to make 12 a constant to clarify the meaning.

Reply:
Done